### PR TITLE
Add documentation for streaming and display related functions 

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**

### CODE REVIEW

File: app/src/handle_swap.h [hunk 1]
```diff
@@ -22,4 +22,9 @@
 
 #pragma once
 
+/**
+ * @brief Called to check the validity of swap params previously communicated
+ * by swap_copy_transaction_parameters . Called from Ledger SDK.
```
Comment (spalmer25) [inline app/src/handle_swap.h:27]: This function is not `Called from Ledger SDK.`

File: app/src/ui_stream.h [hunk 2]
```diff
@@ -124,61 +126,145 @@ typedef uint8_t tz_ui_icon_t;
...
+ * displayed on next screen.)
  */
 size_t tz_ui_stream_push(tz_ui_cb_type_t cb_type, const char *title,
                          const char *value, tz_ui_layout_type_t layout_type,
                          tz_ui_icon_t icon);
+
+/**
+ * @brief  internal implementation of tz_ui_stream_push, implemented
+ * differently for stax and nano* devices.
+ *
```
Comment (spalmer25) [inline app/src/ui_stream.h:216]: The `@params` comments have been forgotten here, particularly for the `max` parameter.

File: app/src/ui_strings.h [hunk 3]
```diff
@@ -16,44 +16,89 @@
...
+ * 'start'.
+ *
+ * @param str pointer to 'start' pointer of the buffer.
+ */
+void ui_strings_drop(char **str);
+/**
+ * @brief Drop the logical last string in the ring buffer, which ends with
+ * 'end' pointer.
+ *
+ * @param str Pointer to pointer of string.
```
Comment (spalmer25) [inline app/src/ui_strings.h:82]:
```suggestion
 * @param str Pointer to 'end' pointer of string.
```

**Review by spalmer25 — COMMENTED** [review]
This MR is more than `Add documentation for ui_stream and ui_string files`.
Could you rename the MR ?

File: app/src/ui_strings.h (hunk 3 above)
Comment (ajinkyaraj-23) [reply app/src/ui_strings.h:82]: It is pointer to start of last string.

File: app/src/ui_stream.h (hunk 2 above)
Comment (ajinkyaraj-23) [reply app/src/ui_stream.h:216]: Added.

File: app/src/handle_swap.h (hunk 1 above)
Comment (ajinkyaraj-23) [reply app/src/handle_swap.h]: changed to 'swap_copy_transaction_parameters which is called from Ledger SDK.'

Comment (spalmer25) [reply app/src/handle_swap.h]: `swap_copy_transaction_parameters` is not called from `Ledger SDK` but from our code, right ?
https://github.com/trilitech/ledger-app-tezos-wallet/blob/main/app/src/apdu_sign.c#L108

Comment (ajinkyaraj-23) [reply app/src/handle_swap.h]: No. We are calling swap_check_validity() from our code. The swap_copy_transaction_parameters is called from main.c :116 in LedgerHQ/ledger-secure-sdk/lib_standard_app

**Pull request merged**